lib/core/queue.js

```diff
@@ -34,6 +34,10 @@ class CommandQueue extends EventEmitter {
     const {compatMode} = this;
     const parentContext = this.currentNode.context;
 
+    if (!this.deferred) {
+      this.deferred = Utils.createPromise();
+    }
```

Comment:
Wouldn't there be a situation where deferred is presend but already resolved?

Reply:
Good catch! You are correct.

Reply:
Ah but it is set to null once the promise has been resolved: https://github.com/nightwatchjs/nightwatch/pull/3822/files/40a2a1146417b6958086e32d0dc0ca97752c89be#diff-76792a675502fe2db89fb60515901ae74edf41d73f32eaeacc6f47d9153654ebL132